.github/workflows/test-snap.yaml

```diff
@@ -0,0 +1,65 @@
+name: Test Snap
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+  # Allow manual trigger
+  workflow_dispatch:
+
+jobs:
+  build:
+    name: Build and test snap
+    strategy:
+      matrix:
+        runs:
+          - architecture: amd64
+            runner: ubuntu-latest
+          - architecture: arm64
+            runner: [self-hosted, ARM64, Linux]
+    runs-on: ${{ matrix.runs.runner }}
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Build snap
+        uses: snapcore/action-build@v1
+        id: snapcraft
+
+      - name: Install and configure snap
+        shell: bash
+        run: |
+          sudo snap install --dangerous ${{ steps.snapcraft.outputs.snap }}
+          sudo snap alias linuxptp.ptp4l ptp4l
+          sudo snap alias linuxptp.hwstamp-ctl hwstamp_ctl
+          sudo snap alias linuxptp.nsm nsm
+          sudo snap alias linuxptp.phc-ctl phc_ctl
+          sudo snap alias linuxptp.phc2sys phc2sys
+          sudo snap alias linuxptp.pmc pmc
+          sudo snap alias linuxptp.timemaster timemaster
+          sudo snap alias linuxptp.ts2phc ts2phc
+          sudo snap alias linuxptp.tz2alt tz2alt
+      
+      - name: Get linuxptp-testsuite
+        shell: bash
+        run: |
+          cd ~
+          git clone https://github.com/mlichvar/linuxptp-testsuite.git
+      
+      - name: Get and build clknetsim
+        shell: bash
+        run: |
+          cd ~/linuxptp-testsuite
+          git clone https://github.com/mlichvar/clknetsim
+          cd clknetsim
+          make
+          # Replace clknetsim's LD_PRELOAD with a copy of the shared object contained inside the snap, using the correct glibc version for the base
+          sed -i -e 's/LD_PRELOAD=${CLKNETSIM_PRELOAD:+$CLKNETSIM_PRELOAD:}$CLKNETSIM_PATH/CLKNETSIM_SO_PATH=\/snap\/linuxptp\/current/g' clknetsim.bash
+
+      - name: Run tests
+        shell: bash
+        run: |
+          cd ~/linuxptp-testsuite
+          ./run
```

Comment:
The steps are still highly coupled (as if they belonged to a single bash script) and a small change will break the subsequent steps without an obvious error. 

Could we use the `checkout` action for the cloning steps (which takes a single commit by default), build, then reorganize files right before running tests?

Reply:
All these could be moved to a script in the source and reused here in a single step, since one needs to run them for manual testing as well.